Mathlib/CategoryTheory/Comma/Basic.lean

```diff
@@ -409,6 +411,37 @@ def post (L : A ⥤ T) (R : B ⥤ T) (F : T ⥤ C) : Comma L R ⥤ Comma (L ⋙
       right := f.right
       w := by simp only [Functor.comp_map, ← F.map_comp, f.w] }
 
+/-- The canonical functor from the product of two categories to the comma category of their
+respective functors into `Discrete PUnit`. -/
+@[simps]
+def PUnitPUnitFromProd (L : A ⥤ Discrete PUnit) (R : B ⥤ Discrete PUnit) :
+    A × B ⥤ Comma L R where
+  obj X :=
+    { left := X.1
+      right := X.2
+      hom := Discrete.eqToHom rfl }
+  map {X} {Y} f :=
+    { left := f.1
+      right := f.2 }
+
+/-- Taking the comma category of two functors into `Discrete PUnit` results in something
+is equivalent to their product. -/
+def PUnitPUnitIso (L : A ⥤ Discrete PUnit) (R : B ⥤ Discrete PUnit) :
+    Comma L R ≌ A × B :=
+  Equivalence.mk ((fst L R).prod' (snd L R)) (PUnitPUnitFromProd L R)
+    { hom := 𝟙 _, inv := 𝟙 _ }
+    { hom := 𝟙 _, inv := 𝟙 _ }
+
+/-- Taking the comma category of a functor into `A ⥤ Discrete PUnit` and the identity
+`Discrete PUnit ⥤ Discrete PUnit` results in a category equivalent to `A`. -/
+def PUnitIdIso (L : A ⥤ Discrete PUnit) : Comma L (Functor.id _) ≌ A :=
```

Comment:
Can you show that the forward direction of this is naturally isomorphic to `Comma.fst`?

Reply:
Also, I guess this statement actually generalized to `Comma L R` for any `R : Discrete PUnit => Discrete PUnit` (which is of course the identity).

Reply:
They're actually equal on the nose to `Comma.fst` and `Comma.snd`